### PR TITLE
Implement metadata

### PR DESCRIPTION
**Pull request by nolway**
Implement the getting of metadata on scripting API.

The metadata is an object generated from the administration. The pusher and the front must be agnostic of the data on this object.

In the SAAS case we send the list of NFTS own by the user.

### CODE REVIEW

File: front/src/Api/Events/ui/UIWebsite.ts [hunk 1]
```diff
@@ -40,8 +40,8 @@ export type UIWebsiteSize = z.infer<typeof isUIWebsiteSize>;
 
 export const isCreateUIWebsiteEvent = z.object({
     url: z.string(),
-    visible: z.optional(z.boolean()),
-    allowApi: z.optional(z.boolean()),
+    visible: z.boolean().optional(),
+    allowApi: z.boolean().optional(),
```
Comment (gparant) [inline front/src/Api/Events/ui/UIWebsite.ts:44]: Why this update and why doesn't apply on all `z.optional(...)`?

File: front/src/Connexion/Room.ts [hunk 2]
```diff
@@ -28,15 +28,15 @@ export class Room {
     private _loadingCowebsiteLogo: string | undefined;
     private _loadingLogo: string | undefined;
     private _loginSceneLogo: string | undefined;
-    //eslint-disable-next-line @typescript-eslint/ban-types
-    private _mucRooms: Array<Object> | undefined;
+    private _metadata: unknown | undefined;
+    private _mucRooms: Array<Record<string, unknown>> | undefined;
```
Comment (gparant) [inline front/src/Connexion/Room.ts:32]: Have you test it on XMPP server?

File: front/src/Connexion/Room.ts [hunk 3]
```diff
@@ -239,8 +240,11 @@ export class Room {
         return this._loginSceneLogo;
     }
 
-    //eslint-disable-next-line @typescript-eslint/ban-types
-    get mucRooms(): Array<Object> | undefined {
+    get metadata(): unknown {
+        return this._metadata;
+    }
+
+    get mucRooms(): Array<Record<string, unknown>> | undefined {
```
Comment (gparant) [inline front/src/Connexion/Room.ts:247]: Same thing, have test it with XMPP server?

File: maps/tests/ApiMetadata/script.js [hunk 4]
```diff
@@ -0,0 +1,3 @@
+WA.onInit().then(() => {
+    console.log("Metadata: ", WA.metadata);
+});
```
Comment (moufmouf) [inline maps/tests/ApiMetadata/script.js:3]: The test map is useless in my opinion.

First, it should have a text associated with it (explaining we need to look at the metadata in the logs)
Second, those tests are run without an Admin API. So WA.metadata will be undefined, unless we specifically require to start with an admin.

I would rather simply modify the `tests/Metadata/getCurrentRoom.json` to add a console log of the metadata with a small comment explaining that metadata are filled only if an admin is started and provides some.

File: front/src/Connexion/Room.ts [hunk 5]
```diff
@@ -239,8 +240,11 @@ export class Room {
         return this._loginSceneLogo;
     }
 
-    //eslint-disable-next-line @typescript-eslint/ban-types
-    get mucRooms(): Array<Object> | undefined {
+    get metadata(): unknown {
```
[the remark this answers is not in the document]
Comment (nolway) [reply front/src/Connexion/Room.ts:243]: Nope because metadata can be undefined

File: front/src/Api/Events/ui/UIWebsite.ts (hunk 1 above)
Comment (nolway) [reply front/src/Api/Events/ui/UIWebsite.ts:44]: The optional wrapper is the legacy syntax and will be deleted in future.

File: messages/JsonMessages/MapDetailsData.ts [hunk 6]
```diff
@@ -81,28 +81,31 @@ export const isMapDetailsData = z.object({
...
-    showPoweredBy: extendApi(z.optional(z.nullable(z.boolean())), {
+    showPoweredBy: extendApi(z.boolean().nullable().optional(), {
         description: "The URL of the image to be used on the name scene",
         example: "https://example.com/logo_login.png",
     }),
-    thirdParty: extendApi(z.optional(z.nullable(isMapThirdPartyData)), {
+    thirdParty: extendApi(isMapThirdPartyData.nullable().optional(), {
         description: "Configuration data for third party services",
     }),
+    metadata: extendApi(z.unknown().optional(), {
```
[the remark this answers is not in the document]
Comment (nolway) [reply messages/JsonMessages/MapDetailsData.ts:106]: Metadata can be undefined

**Pull request merged**